WordPress/Classes/Networking/AccountServiceRemoteREST.m

```diff
@@ -113,6 +117,72 @@ - (void)updateBlogsVisibility:(NSDictionary *)blogs
            }];
 }
 
+- (WordPressComApi *)anonymousApi
+{
+    if (!_anonymousApi) {
+        _anonymousApi = [WordPressComApi anonymousApi];
+    }
+
+    return _anonymousApi;
+}
+
+- (void)findExistingAccountByEmail:(NSString *)email success:(void (^)(BOOL found))success failure:(void (^)(NSError *error))failure
+{
+    // TODO: We need to make a versioned flavor of this endpoint and ensure it always returns a JSON object.
+    // TODO: Remove the special case in `WordPressComApi.assertApiVersion` the endpoint is versioned.
+    NSString *path = @"https://public-api.wordpress.com/is-available/email";
+    [self.api GET:path
+       parameters:@{ @"q": email }
+          success:^(AFHTTPRequestOperation *operation, id responseObject) {
+              BOOL found = NO;
+              if ([responseObject isKindOfClass:[NSDictionary class]]) {
+                  NSDictionary *dict = (NSDictionary *)responseObject;
+                  NSString *errStr = [dict stringForKey:@"error"];
+                  found = [@"taken" isEqualToString:errStr];
+              }
+
+              if (success) {
+                  success(found);
+              }
+          } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+              // The is-available endpoint is an oddball.
```

Comment:
Definitely. Until there's a newer version of this API, I'd add a comment explaining what the possible responses are. This code made no sense until I did a few test requests to that endpoint and saw the responses myself.